mpas_analysis/ocean/ocean_modelvsobs.py

```diff
@@ -94,126 +97,39 @@ def ocn_modelvsobs(config, field):
-                latVarName='lat', lonVarName='lon')
-        if not os.path.exists(regriddedFileName):
-            buildObsClimatologies = True
-            break
-
-    varList = [field]
-
-    if field == 'mld':
-
-        iselvals = None
-
-        obsFieldName = 'mld_dt_mean'
-
-        if buildObsClimatologies:
-            # Load MLD observational data
-            dsObs = xr.open_mfdataset(obsFileName)
-
-            # Increment month value to be consistent with the model output
-            dsObs.iMONTH.values += 1
-
-            # Rename the dimensions to be consistent with other obs. data sets
-            dsObs.rename({'month': 'calmonth', 'lat': 'latCoord',
-                          'lon': 'lonCoord'}, inplace=True)
-            dsObs.rename({'iMONTH': 'Time', 'iLAT': 'lat', 'iLON': 'lon'},
-                         inplace=True)
-            # set the coordinates now that the dimensions have the same names
-            dsObs.coords['lat'] = dsObs['latCoord']
-            dsObs.coords['lon'] = dsObs['lonCoord']
-            dsObs.coords['Time'] = dsObs['calmonth']
-            dsObs.coords['month'] = ('Time', np.array(dsObs['calmonth'], int))
-            # no meaningful year since this is already a climatology
-            dsObs.coords['year'] = ('Time', np.ones(dsObs.dims['Time'], int))
-
-            dsObs = mpas_xarray.subset_variables(dsObs, [obsFieldName,
-                                                         'month'])
-            # Reorder dataset for consistence with other obs. data sets
-            dsObs = dsObs.transpose('Time', 'lat', 'lon')
-
-        # Set appropriate MLD figure labels
-        observationTitleLabel = \
-            "Observations (HolteTalley density threshold MLD)"
-        outFileLabel = "mldHolteTalleyARGO"
-        unitsLabel = 'm'
-
-    elif field == 'sst':
-
-        iselvals = {'nVertLevels': 0}
-
-        climStartYear = config.getint('oceanObservations',
-                                      'sstClimatologyStartYear')
-        climEndYear = config.getint('oceanObservations',
-                                    'sstClimatologyEndYear')
-        timeStart = datetime.datetime(year=climStartYear, month=1, day=1)
-        timeEnd = datetime.datetime(year=climEndYear, month=12, day=31)
-
-        if climStartYear < 1925:
-            period = 'pre-industrial'
-        else:
-            period = 'present-day'
-
-        if buildObsClimatologies:
-            dsObs = xr.open_mfdataset(obsFileName)
-            dsObs.rename({'time': 'Time'}, inplace=True)
-            dsObs = dsObs.transpose('Time', 'lat', 'lon')
-            dsObs = dsObs.sel(Time=slice(timeStart, timeEnd))
-            dsObs.coords['month'] = dsObs['Time.month']
-            dsObs.coords['year'] = dsObs['Time.year']
-
-        obsFieldName = 'SST'
+    setupFuncs = {'mld': _setup_mld,
+                  'sst': _setup_sst,
+                  'sss': _setup_sss}
+    setup = setupFuncs[field]
```

Comment:
This will probably be handled better by having tasks be classes as part of #128.  For now, This dictionary is used to figure out the right setup function for each field.